packages/form-control/README.md

```diff
@@ -0,0 +1,177 @@
+
+  <button type="submit">Submit</button>
+</form>
+
+<script>
+  const form = document.querySelector('form');
+  form.addEventListener('submit', event => {
+    /** Prevent the page from reloading */
+    event.preventDefault();
+
+    /** Get form data object via built-in API */
+    const data = new FormData(event.target);
+    console.log('demo-form-control value:', data.get('demo'));
+  });
+</script>
+```
+
+### ElementInternals
+
+This library makes use of [ElementInternals](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals) features. As of the time of writing `ElementInternals` features are fully supported in Chrome, partially supported in Firefox and being strongly considered by Webkit.
+
+In order to make these features work in all browsers you will need to include the [element-internals-polyfill](https://www.npmjs.com/package/element-internals-polyfill). Refer to the `element-internals-polyfill` documentation for installation and usage instructions.
+
+### `Value` & `checked`
+Any component that uses the `FormControlMixin` will have a `value` property that the element will apply to the parent form. If the element also has a `checked` property on the prototype (think checkbox or radio button) the element's value will only be applied to the parent form when the `checked` property is truthy (like native checkboxes and radio buttons behave)
+
+## Validation
+
+The `FormControlMixin` includes an API for constraint validations and a set of common validators for validity states like `required`, `minlength`, `maxlength` and `pattern`.
+
+```typescript
+import { LitElement, html } from 'lit';
+import { customElement, query, property } from 'lit/decorators.js'
+import { live } from 'lit/directives/live.js';
+
+import { FormControlMixin } from './path/to/mixin';
+import { requiredValidator } from './path/to/mixin/validators';
+
+@customElement('demo-form-control')
+class DemoFormControl extends FormControlMixin(LitElement) {
+  static formControlValidators = [requiredValidator];
+
+  @property({ type: Boolean, reflect: true })
+  required = false;
+
+  @property({ type: String })
+  value = '';
+
+  render() {
+    return html`
+      <label for="input"><slot></slot></label>
+      <input
+        id="input"
+        .value="${live(this.value)}"
+        @input="${this.#onInput}"
+      >
+    `;
+  }
+
+  #onInput({ target }: { target: HTMLInputElement }): void {
+    this.value = target.value;
+  }
+}
+```
+
+Including the `requiredValidator` adds a validation function attached to the `valueMissing` validity state to the component instance.
+
+> Note, this does require the element's prototype to actually have a `required` property defined.
+
+### Validation Target
+Every `FormControlMixin` element will need a public `validationTarget` which must be a focusable DOM element. In the event a control becomes invalid, this item will be focused on form submit for accessibility purposes. Failure to do so will cause an error to throw. 
```

Comment:
Another requirement, it must be in the element's shadow root (not sure if there's a way around this) … 